core/src/main/java/com/flipkart/poseidon/core/PoseidonRequest.java

```diff
@@ -30,14 +30,15 @@
 
     private final Map<String, Object> attributes = new ConcurrentHashMap<>();
     private final String url;
+    private final HttpServletRequest httpRequest;
```

Comment:
Poseidon app can be embedded inside web servlet container or as a stream consumer (ex: kafka consumer). So taking http dependencies here doesn't sound right. This should be generalized to a type T which could be HttpServletRequest for http container, StreamEventRequest for a stream consumer etc.. If that's going to cause backward incompatibility, we can at least make this Optional to denote it is not present always.